### PR TITLE
Prepare switch to GHMQ

### PR DESCRIPTION
**Pull request by eldruin**
Bors has been non-responsive for a few days.

### CODE REVIEW

**Review by Dirbaio — APPROVED** [review]
omg rustup is now broken too? :(

**Comment by adamgreig** [comment]
Could you also delete `bors.toml`?

File: .github/workflows/clippy.yml [hunk 1]
```diff
@@ -1,7 +1,9 @@
 on:
-  push:
-    branches: [ staging, trying, master ]
-  pull_request:
+  push: # Run CI for all branches except GitHub merge queue tmp branches
+    branches-ignore:
+    - "gh-readonly-queue/**"
```
[the remark this answers is not in the document]
Comment (Dirbaio) [reply .github/workflows/clippy.yml:4]: This makes CI *not* run on these branches.

**Comment by Dirbaio** [comment]
> Could you also delete bors.toml?

it's already deleted

File: .github/workflows/clippy.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (adamgreig) [reply .github/workflows/clippy.yml:4]: Right, but why are we telling GHA to not run on that branch?

[the remark this answers is not in the document]
Comment (Dirbaio) [reply .github/workflows/clippy.yml:4]: if you do both "on push to any branch" and "on merge group", when GHMQ pushes to `gh-readonly-queue/blahblah` it matches both which I think triggers duplicate builds.

[the remark this answers is not in the document]
Comment (eldruin) [reply .github/workflows/clippy.yml:4]: Exactly, this avoids duplicate runs. It is taken from [here](https://github.com/tock/tock/pull/3486)

[the remark this answers is not in the document]
Comment (Emilgardis) [reply .github/workflows/clippy.yml:4]: This is only necessary if you don't set branches for the `push` event to work on.

**Pull request merged**